shared/util/tf/tf_generator.go

```diff
@@ -0,0 +1,439 @@
+			}
+
+			serviceConfigs := stackConfig[variable.Name].(map[string]interface{})
+
+			values := map[string]cty.Value{}
+			defaultValues := variable.TypeDefaults.Children[""].DefaultValues
+			for _, service := range tf.happyConfig.GetServices() {
+				elem := map[string]cty.Value{}
+
+				// Sort the service attributes alphabetically
+				attributeNames := reflect.ValueOf(variable.Type.ElementType().AttributeTypes()).MapKeys()
+				sort.SliceStable(attributeNames, func(i, j int) bool {
+					return strings.Compare(attributeNames[i].String(), attributeNames[j].String()) < 0
+				})
+
+				serviceConfig := serviceConfigs[service].(map[string]interface{})
+
+				for i := range attributeNames {
+					k := attributeNames[i].String()
+					if _, ok := elem[k]; !ok {
+						var value cty.Value
+						if configuredValue, ok := serviceConfig[k]; ok {
+							var err error
+							if configuredValue != nil {
+								var valType cty.Type
+								valType, err = gocty.ImpliedType(configuredValue)
+								if err != nil {
+									logrus.Errorf("Unable to determine a parameter (%s) type: %s", k, err.Error())
+								}
+								value, err = gocty.ToCtyValue(configuredValue, valType)
+								if err != nil {
+									logrus.Errorf("Unable to convert a parameter value (%s): %s", k, err.Error())
+								}
+								elem[k] = value
+							}
+						}
+
+						// If default values are known, populate them
+						if value.IsNull() {
+							if defaultValue, ok := defaultValues[k]; ok {
+								if !defaultValue.IsNull() {
+									elem[k] = defaultValue
+								}
+							}
+						}
+					}
+				}
+
+				values[service] = cty.ObjectVal(elem)
+			}
+
+			val := cty.MapVal(values)
+			moduleBlockBody.SetAttributeValue(variable.Name, val)
+		default:
+			var value cty.Value
+			if configuredValue, ok := stackConfig[variable.Name]; ok {
+
+				var err error
+
+				if configuredValue != nil {
+					var valType cty.Type
+					valType, err = gocty.ImpliedType(configuredValue)
+					if err != nil {
+						logrus.Error(err)
+					}
+					value, err = gocty.ToCtyValue(configuredValue, valType)
+					if err != nil {
+						logrus.Error(err)
+					}
+				}
+			}
+
+			if value.IsNull() && !variable.Default.IsNull() {
+				value = variable.Default
+			}
+
+			moduleBlockBody.SetAttributeValue(variable.Name, value)
+		}
+	}
+
+	_, err = tfFile.Write(hclFile.Bytes())
+
+	return err
+}
+
+func (tf *TfGenerator) GenerateProviders(srcDir string) error {
+	tfFile, err := os.Create(filepath.Join(srcDir, "providers.tf"))
+	if err != nil {
+		return errors.Wrap(err, "Unable to generate HCL code")
+	}
+	defer tfFile.Close()
+	hclFile := hclwrite.NewEmptyFile()
+
+	rootBody := hclFile.Body()
+	err = tf.generateAwsProvider(rootBody, "", "${var.aws_account_id}", "${var.aws_role}")
+	if err != nil {
+		return errors.Wrap(err, "Unable to generate HCL code for AWS provider")
+	}
+
+	err = tf.generateAwsProvider(rootBody, "czi-si", "626314663667", "tfe-si")
+	if err != nil {
```

Comment:
right now the provider.tf is a static file. I think we should use a template for it since there is no dynamic configuration.

Reply:
Region is dynamic.